hatch/src/cli/commands/new.rs

```diff
@@ -42,16 +42,13 @@ impl<'new> New {
   }
 
   fn construct_deps_string(&self, deps: &Vec<(String, String)>) -> String {
-    deps.iter().map(|&(ref url, ref name)| {
-      let mut includes = String::new();
-      includes.push_str("deps:\n");
-      includes.push_str("  ");
-      includes.push_str(&name[..]);
-      includes.push_str(": ");
-      includes.push_str(&url[..]);
-      includes.push_str("\n");
-      includes
-    }).collect()
+    if deps.len() > 0 {
+      String::from("deps:\n") + deps.iter().map(|&(ref url, ref name)| {
+        format!("  {}: {}\n", &name[..], &url[..])
```

Comment:
what is this syntax?

Reply:
It produces a deps section with N dependencies greater than or equal to 1 or no deps section at all.